cmd/boostd/backup.go

```diff
@@ -0,0 +1,274 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"io/ioutil"
+	"os"
+	"path"
+	"time"
+
+	"github.com/mitchellh/go-homedir"
+	"github.com/urfave/cli/v2"
+	"gopkg.in/cheggaaa/pb.v1"
+
+	"github.com/filecoin-project/boost/node"
+	"github.com/filecoin-project/lotus/chain/store"
+	"github.com/filecoin-project/lotus/lib/backupds"
+	lotus_repo "github.com/filecoin-project/lotus/node/repo"
+	dstore "github.com/ipfs/go-datastore"
+)
+
+var fm = map[string]fs.FileMode{"api": 0644, "boost.db": 0644, "boost.logs.db": 0644, "config.toml": 0664, "storage.json": 0644, "token": 0600}
+
+var backupCmd = &cli.Command{
+	Name:  "backup",
+	Usage: "Perorms offline backup of the Boost subsystem",
+	Flags: []cli.Flag{
+		&cli.PathFlag{
+			Name:     "path",
+			Usage:    "Specify a directory to create backup. Must be inside home directory.",
```

Comment:
Why does it have to be inside the home directory?

Reply:
I think people will quite often want to backup files to somewhere other than their homedir (eg to an external drive).
We can use it up to the user to figure out user permissions.

Reply:
It will work outside of home directory as well. If a user providers an absolute path, back would be done there.